cms/core/db_router.py

```diff
@@ -0,0 +1,55 @@
+from typing import Any, Optional
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.db import DEFAULT_DB_ALIAS, router, transaction
+from django.db.models import Model
+
+
+class ReadReplicaRouter:  # pylint: disable=unused-argument,protected-access
+    """A database router which routes read queries to a read replica.
+
+    https://docs.djangoproject.com/en/5.1/topics/db/multi-db/#automatic-database-routing
+    """
+
+    REPLICA_DB_ALIAS = "read_replica"
+
+    REPLICA_DBS = frozenset({REPLICA_DB_ALIAS, DEFAULT_DB_ALIAS})
+
+    def __init__(self) -> None:
+        if self.REPLICA_DB_ALIAS not in settings.DATABASES:  # pragma: no cover
+            raise ImproperlyConfigured("Read replica is not configured.")
+
+    def db_for_read(self, model: type[Model], **hints: Any) -> Optional[str]:
+        """Which database should be used for read queries?"""
```

Comment:
Let's make the doc strings a bit clearer.
```suggestion
        """Determines which database should be used for read operations.

        By default, read queries are routed to the read replica. However, if a transaction
        is active on the default database, reads are directed to the default database to
        ensure consistency.
```

Reply:
Yup, that's fine, but let's not use 'questions' for docstring as it won't be clearer to all levels. It should be descriptive or imperative style. So in this cause I would still suggest we use `Determines which database should be used for read operations.`